src/explore-education-statistics-frontend/src/middleware/pages/redirectPages.ts

```diff
@@ -0,0 +1,62 @@
+import redirectService, {
+  Redirects,
+  RedirectType,
+} from '@frontend/services/redirectService';
+import type { NextRequest } from 'next/server';
+import { NextResponse } from 'next/server';
+
+interface StoredRedirects {
+  redirects: Redirects;
+  fetchedAt: number;
+}
+
+let storedRedirects: StoredRedirects | undefined;
+
+const redirectPaths = {
+  methodologyRedirects: '/methodology',
+  publicationRedirects: '/find-statistics',
+};
+
+const cacheTime = 300000;
+
+export default async function redirectPages(request: NextRequest) {
+  const shouldRefetch =
+    !storedRedirects ||
+    (storedRedirects?.fetchedAt &&
+      storedRedirects.fetchedAt + cacheTime < Date.now());
+
+  if (shouldRefetch) {
+    storedRedirects = {
+      redirects: await redirectService.list(),
+      fetchedAt: Date.now(),
+    };
+  }
+
+  let redirectUrl: string | undefined;
+
+  Object.keys(redirectPaths).forEach(key => {
+    if (
+      request.nextUrl.pathname.startsWith(redirectPaths[key as RedirectType])
```

Comment:
Could extract `key` to a `redirectType` variable to avoid doing `as RedirectType` twice